fixtures/pages-workerjs-and-functions-app/tests/index.test.ts

```diff
@@ -73,6 +44,40 @@ describe("Pages project with `_worker.js` and `/functions` directory", () => {
 			response = await fetch(`http://${ip}:${port}/party`);
 			text = await response.text();
 			expect(text).toEqual("Oops! Tous les alligators sont allés à la fête 🎉");
-		}
+			await exit();
+		},
+		10000
 	);
 });
+
+async function runWranglerPagesDev(publicPath: string, ...options: string[]) {
+	let resolveReadyPromise: (value: { ip: string; port: number }) => void;
+	const ready = new Promise<{ ip: string; port: number }>(
+		(resolve) => (resolveReadyPromise = resolve)
+	);
+	const wranglerProcess = fork(
+		"../../packages/wrangler/bin/wrangler.js",
+		["pages", "dev", publicPath, ...options],
+		{
+			stdio: ["ignore", "ignore", "ignore", "ipc"],
+			cwd: path.resolve(__dirname, ".."),
+		}
+	).on("message", (message) => {
+		resolveReadyPromise(JSON.parse(message.toString()));
+	});
+	async function exit() {
+		await ready;
+		return new Promise((resolve, reject) => {
+			wranglerProcess.once("exit", (code) => {
+				if (!code) {
+					resolve(code);
+				} else {
+					reject(code);
+				}
+			});
+			wranglerProcess.kill("SIGTERM");
+		});
+	}
+	const { ip, port } = await ready;
+	return { ip, port, exit };
+}
```

Comment:
This looks pretty similar to the previous definition—could this be lifted out into a helper file somewhere?

Reply:
LOL! Yes I totally agree (and just said so in a previous comment before I read this).
I was not 100% sure where shared code between tests in different fixtures should go though...

Reply:
Yeah, that's fair. A bunch of the fixtures should probably be changed once @mrbbot's e2e testing has landed, as well, so this can be a problem for the future